syntax = "proto3"; // Defino la sintaxis del archivo proto (proto3 es la sintaxis mas actual)

package stockearte; // Defino el nombre del paquete

service Stockearte // Creo el servicio y los métodos que va a tener
{ 
    // PUNTO 1.A
    rpc altaTienda (altaTiendaRequest) returns (mensajeSimple) {} // Defino un método unario
    rpc bajaTienda (bajaTiendaRequest) returns (mensajeSimple) {}

    // PUNTO 1.B
    rpc altaUsuario (altaUsuarioRequest) returns (mensajeSimple) {}

    // PUNTO 1.C
    rpc altaProducto (altaProductoRequest) returns (mensajeSimple) {}

    // PUNTO 1.D
    rpc modificacionProducto (modificacionProductoRequest) returns (mensajeSimple) {}
}

/*********************************************************************************************/
// DEFINICIÓN DE TIPOS DE DATOS

message mensajeSimple
{
    string mensaje = 1;
}


// PUNTO 1.A
// Servicio altaTienda
message altaTiendaRequest     // Defino que datos va a tener el tipo de dato altaTiendaRequest
{   
    string codigoTienda = 1;  // Las variables deben estar ordenadas numericamente
    string direccion    = 2;
    string ciudad       = 3;
    string provincia    = 4;
    bool   habilitado   = 5;
}

// Servicio bajaTienda
message bajaTiendaRequest 
{
    string codigoTienda = 1;
}


// PUNTO 1.B
// Servicio altaUsuario
message altaUsuarioRequest
{
    string nombre        = 1;
    string apellido      = 2;
    string nombreUsuario = 3;
    string contrasenia   = 4;
    bool   habilitado    = 5;
    string codigoTienda  = 6; 
}


// PUNTO 1.C
// Servicio altaProducto
message altaProductoRequest
{
    string nombre         = 1;
    string codigoProducto = 2;
    string talle          = 3;
    string foto           = 4;
    string color          = 5;
    int32  stock          = 6;
}


// PUNTO 1.D
// Servicio modificacionProducto
message modificacionProductoRequest
{
    string codigoProducto = 1;
    int32  nuevoStock     = 2;
}